src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -37,6 +37,16 @@ namespace System.Net.Quic;
 /// </remarks>
 public sealed partial class QuicConnection : IAsyncDisposable
 {
+#if DEBUG
+    /// <summary>
+    /// Name of file where secrets should be stored.
+    ///</summary>
+    private static readonly string? KeyLogFile = Environment.GetEnvironmentVariable("SSLKEYLOGFILE");
```

Comment:
Would it make sense to collapse the logic around `KeyLogFile` into `MsQuicTlsSecret` so it's all in one place?

Reply:
I would not mind but I'm not sure how you propose to do.  `KeyLogFile` is the trigger to enable it. I did not want to dereference `_tlsSecret` just because the `KeyLogFile` has value and the `_tlsSecret` needs to live per instance.

Reply:
I was thinking of static Create method that could return null if the variable is not set. There wouldn't be any unnecessary allocation, the Write could be just `_tlsSecret?.Write...`. ` KeyLogFile` would remain static just inside `MsQuicTlsSecret`. WDYT?

Reply:
fine with me. It allows to encapsulate more in `MsQuicTlsSecret`.

Reply:
updated.